src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -847,5 +848,95 @@ public static BinderCheckTypeResult CheckBinaryOpCore(IErrorContainer errorConta
+        {
+            Contracts.AssertValue(node);
+            nodeValue = null;
+            switch (node.Kind)
+            {
+                case NodeKind.StrLit:
+                    nodeValue = node.AsStrLit().Value;
+                    return true;
+                case NodeKind.BinaryOp:
+                    var binaryOpNode = node.AsBinaryOp();
+                    if (binaryOpNode.Op == BinaryOp.Concat)
+                    {
+                        if (TryGetConstantValue(context, binaryOpNode.Left, out var left) && TryGetConstantValue(context, binaryOpNode.Right, out var right))
+                        {
+                            nodeValue = string.Concat(left, right);
+                            return true;
+                        }
+                    }
+
+                    break;
+                case NodeKind.Call:
+                    var callNode = node.AsCall();
+                    if (callNode.Head.Name.Value == BuiltinFunctionsCore.Concatenate.Name)
+                    {
+                        var parameters = new List<string>();
+                        foreach (var argNode in callNode.Args.Children)
+                        {
+                            if (TryGetConstantValue(context, argNode, out var argValue))
+                            {
+                                parameters.Add(argValue);
+                            }
+                            else
+                            {
+                                break;
+                            }
+                        }
+
+                        if (parameters.Count == callNode.Args.Count)
+                        {
+                            nodeValue = string.Join(string.Empty, parameters);
+                            return true;
+                        }
+                    }
+
+                    break;
+                case NodeKind.FirstName:
+                    // Possibly a non-qualified enum value
+                    var firstNameNode = node.AsFirstName();
+
+                    if (context.NameResolver.Lookup(firstNameNode.Ident.Name, out var firstNameInfo, NameLookupPreferences.None))
+                    {
+                        if (firstNameInfo.Kind == BindKind.Enum)
+                        {
+                            if (firstNameInfo.Data is string enumValue)
+                            {
+                                nodeValue = enumValue;
+                                return true;
+                            }
+                        }
+                    }
+
+                    break;
+                case NodeKind.DottedName:
+                    // Possibly an enumeration
+                    var dottedNameNode = node.AsDottedName();
+                    if (dottedNameNode.Left.Kind == NodeKind.FirstName)
+                    {
+                        if (context.NameResolver.Lookup(dottedNameNode.Left.AsFirstName().Ident.Name, out var dottedNameInfo, NameLookupPreferences.None))
```

Comment:
>context.NameResolver.Lookup [](http://example.com/codeflow?start=28&length=27)

This was changed from:
if (Document.GlobalScope.TryGetNamedEnum(dottedNameNode.Left.AsFirstName().Ident.Name, out var enumType))

Reply:
This change will be in a future PR.